activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -90,7 +90,18 @@ def alias_attribute_method_definition(code_generator, pattern, new_name, old_nam
         manually_defined = method_defined && self.instance_method(target_name).owner != generated_attribute_methods
         reserved_method_name = ::ActiveRecord::AttributeMethods.dangerous_attribute_methods.include?(target_name)
 
-        if manually_defined && !reserved_method_name
+        if !has_attribute?(old_name)
```

Comment:
In an unrelated conversation I realized that this is something that will disallow alias attribute definitions in an abstract class. For example in an inheritance chain like `MyModel < AbstractClassA < AbstractClassB < ApplicationRecord < ActiveRecord::Base` `alias_attribute` defined in abstract classes will most likely raise a deprecation or even fail since these abstract classes have no attributes while it used to be possible to define alias attributes on abstract classes and it was a valid use-case

So here we should most likely ignore if the definition is happening on an abstract class as we can't know if the attribute exists on a concrete class inherited from the abstract class

Reply:
Nice, thanks for catching that. I'll add a test for that case and add that fix.

Reply:
I incorporated the changes from `main`, which included https://github.com/rails/rails/pull/48991, and added the check for the `abstract_class`, but it looks like this change [broke some other behavior](https://buildkite.com/rails/rails/builds/98910#018a198e-03ba-4086-9e0a-4cd7e66189b0) in a different test file. I'm going to investigate why this broke...